### DISCUSSION TITLE
Deleting and creating rows with foreign keys via the api

### DISCUSSION THREAD

**Discussion by hugolardosa** [nocodb/nocodb#3930] [Q&A]
I’m currently developing a javascript module that automates reads and writes from nocodb, I’m using the nocodb sdk and checking the API rules. 
I am having trouble understanding how I can delete rows that contain links to another rows (foreign keys), is there any parameter that can be passed trought the API call, or can I delete the all links via the API and then ask to remove the row?

Another trouble I’m having is creating a row where a cell has links to another rows, I use the rows IDs but I can’t link them, is there a specific call to make that I might be missing?

**Comment by pranavxc** [comment]
You can iterate over the LTAR columns and unlink the records by using the following API - https://apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-nested-remove , after removing all the links you can delete the record.


Same way for linking the record you can use the following - https://apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-nested-add 


If you are using SDK then you can refer to the `link` and `unlink` method in the following composable - https://github.com/nocodb/nocodb/blob/develop/packages/nc-gui/composables/useLTARStore.ts#L238
